### PR TITLE
[Improvement] Add RPC metrics

### LINKED ISSUE #16119: [Improvement] Add metrics for RPC module
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Expose metrics for rpc, help to monitor the rpc error/ time cost.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Purpose of the pull request

close https://github.com/apache/dolphinscheduler/issues/16119

## Brief change log

- Change default sync request timeout to 10s
- Set heartbeat interval to 10s, connectionIdleTime time to 60s
- Expose metrics for rpc exception/cost

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-extract/dolphinscheduler-extract-base/src/test/java/org/apache/dolphinscheduler/extract/base/metrics/RpcMetricsTest.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import org.junit.jupiter.api.Test;
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
+class RpcMetricsTest {
+
+    @Test
+    public void setup() {
+        Metrics.globalRegistry.clear();
+        // 添加一个简单的内存中指标注册表
```
Comment (SbloodyS) [inline dolphinscheduler-extract/dolphinscheduler-extract-base/src/test/java/org/apache/dolphinscheduler/extract/base/metrics/RpcMetricsTest.java]: We should avoid chinese...

**Review by SbloodyS — CHANGES_REQUESTED** [review]

**Review by rickchengx — DISMISSED** [review]
LGTM

File: dolphinscheduler-extract/dolphinscheduler-extract-base/src/test/java/org/apache/dolphinscheduler/extract/base/metrics/RpcMetricsTest.java (hunk 1 above)
Comment (ruanwenjun) [reply dolphinscheduler-extract/dolphinscheduler-extract-base/src/test/java/org/apache/dolphinscheduler/extract/base/metrics/RpcMetricsTest.java]: Done

**Review by SbloodyS — COMMENTED** [review]
CI did not passed.

**Review by SbloodyS — DISMISSED** [review]
+1

**Review by SbloodyS — APPROVED** [review]

**Review by rickchengx — APPROVED** [review]
+1

**Pull request merged**